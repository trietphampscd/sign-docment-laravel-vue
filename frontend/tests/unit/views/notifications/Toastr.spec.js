import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Toastr from '@/views/notifications/Toastr'

Vue.use(BootstrapVue)

describe('Toastr.vue', () => {
  it('has a name', () => {
    expect(Toastr.name).toMatch('toastr')
  })
  it('has a created hook', () => {
    expect(typeof Toastr.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Toastr)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Toastr', () => {
    const wrapper = shallowMount(Toastr)
    expect(wrapper.is(Toastr)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Toastr)
    expect(wrapper.find('div.card-header > div').text()).toMatch('Toastr')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Toastr)
    expect(wrapper.element).toMatchSnapshot()
  })
})
