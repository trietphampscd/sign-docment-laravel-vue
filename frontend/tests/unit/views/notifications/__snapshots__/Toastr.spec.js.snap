// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toastr.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <bcard-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    show-footer=""
    subtitletag="h6"
    tag="div"
    titletag="h4"
  >
    <div
      slot="header"
    >
      <i
        class="icon-info"
      />
       Toastr
      
      <a
        class="badge badge-danger ml-1"
        href="https://coreui.io/pro/vue/"
        rel="noreferrer noopener"
        target="_blank"
      >
        CoreUI Pro
      </a>
       
      <div
        class="card-header-actions"
      >
        <a
          class="card-header-action"
          href="https://github.com/se-panfilov/vue-notifications"
          rel="noreferrer noopener"
          target="_blank"
        >
          <small
            class="text-muted"
          >
            docs
          </small>
        </a>
      </div>
    </div>
     
    <bbutton-stub
      activeclass="active"
      class="mr-1"
      event="click"
      exactactiveclass="active"
      routertag="a"
      target="_self"
      type="button"
      variant="success"
    >
      Success
    </bbutton-stub>
     
    <bbutton-stub
      activeclass="active"
      class="mr-1"
      event="click"
      exactactiveclass="active"
      routertag="a"
      target="_self"
      type="button"
      variant="info"
    >
      Info
    </bbutton-stub>
     
    <bbutton-stub
      activeclass="active"
      class="mr-1"
      event="click"
      exactactiveclass="active"
      routertag="a"
      target="_self"
      type="button"
      variant="warning"
    >
      Warning
    </bbutton-stub>
     
    <bbutton-stub
      activeclass="active"
      class="mr-1"
      event="click"
      exactactiveclass="active"
      routertag="a"
      target="_self"
      type="button"
      variant="danger"
    >
      Error
    </bbutton-stub>
  </bcard-stub>
</div>
`;
